dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/BNFHeadersImpl.java

```diff
@@ -1027,7 +1060,8 @@ public boolean containsHeader(HeaderKeys key) {
         if (null == key) {
             throw new IllegalArgumentException("Null input provided");
         }
-        return (null != findHeader(key));
+        int ord = key.getOrdinal();
+        return (ord <= maxHeadersAdded) ? (headersAdded.get(ord) && (null != findHeader(key))) : (null != findHeader(key));
```

Comment:
Can maybe simplify this as:
```suggestion
        if (ord <= maxHeadersAdded && !headersAdded.get(ord)) {
            return false;
        }
        return (null != findHeader(key);
```

Reply:
Nice - done